pkgs/by-name/ed/edopro/package.nix

```
@@ -0,0 +1,320 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchzip,
+  premake5,
+  writeShellScriptBin,
+  runCommandLocal,
+  symlinkJoin,
+  imagemagick,
+  bzip2,
+  curl,
+  flac,
+  # Use fmt 10+ after release 40.1.4+
+  fmt_9,
+  freetype,
+  libevent,
+  libgit2,
+  libGL,
+  libGLU,
+  libjpeg,
+  libpng,
+  libvorbis,
+  libX11,
+  libxkbcommon,
+  libXxf86vm,
+  lua5_3,
+  mono,
+  nlohmann_json,
+  openal,
+  SDL2,
+  sqlite,
+  wayland,
+  egl-wayland,
+  covers_url ? "https://pics.projectignis.org:2096/pics/cover/{}.jpg",
+  fields_url ? "https://pics.projectignis.org:2096/pics/field/{}.png",
+  # While ygoprodeck has higher quality images, "spamming" of their api results in a ban.
+  # Thats why this link can change since it's compiled into the program, However it will
+  # download assets when needed so it is unlikely to get banned. Unless you search the
+  # card list with no filters of any kind. When testing use ProjectIgnis' website instead.
+  pics_url ? "https://images.ygoprodeck.com/images/cards/{}.jpg",
+}:
+let
+  archLabel =
+    {
+      "x86_64-linux" = "x64";
+      "aarch64-linux" = "arm64";
+    }
+    .${stdenv.hostPlatform.system}
+      or (throw "Don't know how to translate ${stdenv.hostPlatform.system} into an arch label for edopro");
+
+  deps = import ./deps.nix;
+in
+let
+  assets = {
+    pname = "assets";
+    version = deps.edopro-version;
+
+    src = fetchzip {
+      url = "https://github.com/ProjectIgnis/edopro-assets/releases/download/${deps.edopro-version}/ProjectIgnis-EDOPro-${deps.edopro-version}-linux.tar.gz";
+      sha256 = deps.assets-hash;
+    };
+  };
```

Just this should be enough I think.

```suggestion
  assets = fetchzip {
    url = "https://github.com/ProjectIgnis/edopro-assets/releases/download/${deps.edopro-version}/ProjectIgnis-EDOPro-${deps.edopro-version}-linux.tar.gz";
    sha256 = deps.assets-hash;
  };
```

---

I don't see what's changed?

---

```nix
assets = {
  pname = "...";
  version = "...";

  src = fetchzip { /* ... */ };
};
```
->
```nix
assets = fetchzip { /* ... */ };
```